tests/language_specific/chinese/test_chinese_language_generator.py

```diff
@@ -1702,8 +2221,72 @@ def test_roll():
     assert generated_tokens(situation) == (
         "bau3 bau3",
         "dzai4",
-        "syang1",
+        "syang1 dz",
         "shang4",
         "gwun3",
         "chyu1 chi2 bing3",
     )
+
+
+"""TESTS FOR X_IS_Y"""
+
+# simple x_is_y
+def test_the_ball_is_green():
+    ball = situation_object(BALL, properties=[GREEN])
+    situation = HighLevelSemanticsSituation(
+        ontology=GAILA_PHASE_1_ONTOLOGY,
+        salient_objects=[ball],
+        syntax_hints=[ATTRIBUTES_AS_X_IS_Y],
+    )
+    assert generated_tokens(situation) == ("chyou2", "shr4", "lyu4 se4")
+
+
+# x_is_y with first person possession
+def test_my_ball_is_green():
+    ball = situation_object(BALL, properties=[GREEN])
+    dad = situation_object(DAD, properties=[IS_SPEAKER])
+    situation = HighLevelSemanticsSituation(
+        ontology=GAILA_PHASE_1_ONTOLOGY,
+        salient_objects=[ball, dad],
+        always_relations=[Relation(HAS, dad, ball)],
+        syntax_hints=[IGNORE_HAS_AS_VERB, ATTRIBUTES_AS_X_IS_Y],
+    )
+    assert generated_tokens(situation) == ("wo3", "de", "chyou2", "shr4", "lyu4 se4")
+
+
+# x_is_y with second person possession
+def test_your_ball_is_red():
+    ball = situation_object(BALL, properties=[RED])
+    dad = situation_object(DAD, properties=[IS_ADDRESSEE])
+    situation = HighLevelSemanticsSituation(
+        ontology=GAILA_PHASE_1_ONTOLOGY,
+        salient_objects=[ball, dad],
+        always_relations=[Relation(HAS, dad, ball)],
+        syntax_hints=[IGNORE_HAS_AS_VERB, ATTRIBUTES_AS_X_IS_Y],
+    )
+    assert generated_tokens(situation) == ("ni3", "de", "chyou2", "shr4", "hung2 se4")
+
+
+# x_is_y for classifier sentence
+def test_two_balls_are_green():
+    ball1 = situation_object(BALL, properties=[GREEN], debug_handle="ball1")
+    ball2 = situation_object(BALL, properties=[GREEN], debug_handle="ball2")
```

Comment:
Is there a reason you need to specify the debug handle here? My guess is because the default is simply "ball"

Reply:
Yes, it's so if something goes wrong they can be distinguished to determine where the error was. I can take them out now they're passing if you think that's a good idea.

Reply:
No, if that's all it is it's fine to leave in. Helpful if someone accidentally breaks the tests in the future. I was just under the impression we had tried to automatically handle distinguishing between multiple of the same ontology node, but I may be mistaken for this implementation.